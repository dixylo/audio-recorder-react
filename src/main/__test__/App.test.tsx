import { fireEvent, render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import App from '..';

describe('Audio Recorder App', () => {
  let windowSpy: any;

  beforeEach(() => {
    windowSpy = jest.spyOn(window, 'window', 'get');
    render(<App />);
  });
  
  afterEach(() => {
    windowSpy.mockRestore();
  });

  it('should render correctly', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render a header', () => {
    const header = screen.getByText(/Audio Recorder/i);
    expect(header).toBeInTheDocument();
  });

  it('should render a Record button', () => {
    const recordButton = screen.getByTestId('btn-Record');
    expect(recordButton).toBeInTheDocument();
  });

  it('should render a Stop button', () => {
    const stopButton = screen.getByTestId('btn-Stop');
    expect(stopButton).toBeInTheDocument();
  });

  it('should render an instruction', () => {
    const instruction = screen.getByText(/Please click/i);
    expect(instruction).toBeInTheDocument();
  });

  it('should render an audio player', () => {
    const audioplayer = screen.getByTestId('audioplayer');
    expect(audioplayer).toBeInTheDocument();
  });

  it('should render a disabled Stop button', () => {
    const stopButton = screen.getByTestId('btn-Stop');
    expect(stopButton).toBeDisabled();
  });

  it('should render a hidden audio player', () => {
    const audioplayer = screen.getByTestId('audioplayer');
    expect(audioplayer).not.toBeVisible();
  });

  it('should disable Record button on Record button clicked', () => {
    windowSpy.mockImplementation(() => {
      const recordButton = screen.getByTestId('btn-Record');
      fireEvent.click(recordButton);
      expect(recordButton).toBeDisabled();
    });
  });

  it('should enable Stop button on Record button clicked', () => {
    windowSpy.mockImplementation(() => {
      const recordButton = screen.getByTestId('btn-Record');
      const stopButton = screen.getByTestId('btn-Stop');
      fireEvent.click(recordButton);
      expect(stopButton).toBeEnabled();
    });
  });

  it('should disable Stop button on Stop button clicked', () => {
    windowSpy.mockImplementation(() => {
      const recordButton = screen.getByTestId('btn-Record');
      const stopButton = screen.getByTestId('btn-Stop');
      fireEvent.click(recordButton);
      fireEvent.click(stopButton)
      expect(stopButton).toBeDisabled();
    });
  });

  it('should show the audio player on Stop button clicked', () => {
    windowSpy.mockImplementation(() => {
      const recordButton = screen.getByTestId('btn-Record');
      const stopButton = screen.getByTestId('btn-Stop');
      const audioplayer = screen.getByTestId('audioplayer');
      fireEvent.click(recordButton);
      fireEvent.click(stopButton)
      expect(audioplayer).toBeVisible();
    });
  });

  it('should enable Record button on Stop button clicked', () => {
    windowSpy.mockImplementation(() => {
      const recordButton = screen.getByTestId('btn-Record');
      const stopButton = screen.getByTestId('btn-Stop');
      fireEvent.click(recordButton);
      fireEvent.click(stopButton)
      expect(recordButton).toBeEnabled();
    });
  });


});
