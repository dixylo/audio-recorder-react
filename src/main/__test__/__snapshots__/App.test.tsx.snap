// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Audio Recorder App should render correctly 1`] = `
<div
  className="container"
>
  <div
    className="sub-container"
  >
    <h2
      className="header"
    >
      Audio Recorder
    </h2>
    <div
      className="button-container"
    >
      <button
        className="button"
        data-testid="btn-Record"
        disabled={false}
        onClick={[Function]}
      >
        Record
      </button>
      <button
        className="button"
        data-testid="btn-Stop"
        disabled={true}
        onClick={[Function]}
      >
        Stop
      </button>
    </div>
    <p
      className="instruction"
      data-testid="instruction"
    >
      Please click "Record" to start recording.
    </p>
    <audio
      controls={true}
      data-testid="audioplayer"
      src=""
      style={
        Object {
          "visibility": "hidden",
        }
      }
    >
      Your browser does not support the audio element.
    </audio>
  </div>
</div>
`;
